Run local tests with run-test262

bnoordhuis opened the pull request:
First commit is #579.

saghul commented:
Love this!

saghul reviewed (approved):
LGTM! How do you want to proceed? Land this and fix run-262 later so we can run the actual suite on Windows or the other way around?

bnoordhuis commented:
I'll go ahead and land this because this whole exercise ultimately is so I can add an expect-fail test for #567 :-)

bnoordhuis committed:
Run local tests with run-test262

bnoordhuis committed:
Build and run run-test262 on Windows

The ulterior motive here is not that I want to increase CI times
further but that I want to repurpose run-test262 for running our
own tests.

bnoordhuis committed:
Remove obsolete run-test262 style option

We always run the test262 suite in new-style mode.
